incubator/hnc/pkg/reconcilers/hnc_config_test.go

```diff
@@ -324,6 +324,42 @@ var _ = Describe("HNCConfiguration", func() {
 
 		Eventually(getNumPropagatedObjects(ctx, "v1", "LimitRange"), statusUpdateTime).Should(Equal(0))
 	})
+
+	It("should set NumSourceObjects for a type in propagate mode", func() {
+		addToHNCConfig(ctx, "v1", "LimitRange", api.Propagate)
+		makeObject(ctx, "LimitRange", fooName, "foo-lr")
+
+		// There should be 2 source objects of LimitRange: 1 added in this test case, the other added in the previous test case.
```

Comment:
Tests shouldn't depend on each other, I don't think we even guarantee the order in which tests are run. Is there a way we can do cleanups? E.g. `makeObject` stores each object in a map, and then `AfterEach` (at the top of this file) deletes them?

Reply:
Good idea! Thanks for the suggestion! I manually deleted those objects for the `NumSourceObjects` test cases. I added a TODO and fired an issue (#586) for this. I will address this issue in next PR just in case I need to change other test cases.

Reply:
sg